### DISCUSSION TITLE
Apache 403 Forbidden

### DISCUSSION THREAD

**Discussion by ukienitz** [ltb-project/self-service-password#621] [Q&A]
I have the 403 failure! All configurations are tested - please help!

**Comment by coudot** [comment]
Please provide more information on your problem, we are not psychic readers

**Reply by ukienitz** [reply]
I'm sorry, you are right!
Here now more detailed information about the problem:
For many months I have been running the application under centOS 7 with Apache web server without any problems. Since a few weeks I get the Forbidden message when calling the application. There is my webserver configuration file:
```
Alias /ssp /usr/share/self-service-password
<Directory /usr/share/self-service-password>
AllowOverride None
Require all granted
DirectoryIndex index.php
AddDefaultCharset UTF-8

<Directory /usr/share/self-service-password/scripts>
AllowOverride None
Require all denied
Order Deny,Allow
Deny from all
DirectoryIndex index.php
AddDefaultCharset UTF-8
```
Path of the application on the server:
```
╰─ ls -al /usr/share/self-service-password ─╯
Total 8
drwxr-xr-x 8 root root 87 Jan 6 16:47 .
drwxr-xr-x. 120 root root 4096 Jan 7 12:25 ...
drwxr-xr-x 2 root root 28 Jan 6 16:47 conf
drwxr-xr-x 6 root root 249 Jan 6 16:47 htdocs
drwxr-xr-x 2 root root 4096 Jan 6 16:47 lang
drwxr-xr-x 3 root root 160 Jan 6 16:47 lib
drwxr-xr-x 2 root root 94 Jan 6 16:47 scripts
drwxr-xr-x 2 root root 270 Jan 6 16:47 templates
```
The server is needed urgently, so I would be very glad if you could help me!
The URL is:  https://cloud.digital-hub-luebeck.de/ssp/
Thanks a lot!

**Comment by ukienitz** [comment]
I'm sorry, you are right! 
Here now more detailed information about the problem:
For many months I have been running the application under centOS 7 with Apache web server without any problems. Since a few weeks I get the Forbidden message when calling the application. There is my webserver configuration file:

Alias /ssp /usr/share/self-service-password
<Directory /usr/share/self-service-password>
           AllowOverride None
           Require all granted
           DirectoryIndex index.php
           AddDefaultCharset UTF-8
</directory>
<Directory /usr/share/self-service-password/scripts>
           AllowOverride None
           Require all denied
           Order Deny,Allow
           Deny from all
           DirectoryIndex index.php
           AddDefaultCharset UTF-8
</directory>

Path of the application on the server:

╰─ ls -al /usr/share/self-service-password ─╯
Total 8
drwxr-xr-x 8 root root 87 Jan 6 16:47 .
drwxr-xr-x. 120 root root 4096 Jan 7 12:25 ...
drwxr-xr-x 2 root root 28 Jan 6 16:47 conf
drwxr-xr-x 6 root root 249 Jan 6 16:47 htdocs
drwxr-xr-x 2 root root 4096 Jan 6 16:47 lang
drwxr-xr-x 3 root root 160 Jan 6 16:47 lib
drwxr-xr-x 2 root root 94 Jan 6 16:47 scripts
drwxr-xr-x 2 root root 270 Jan 6 16:47 templates

The server is needed urgently, so I would be very glad if you could help me!

Thanks a lot!


Translated with www.DeepL.com/Translator (free version)

**Comment by coudot** [comment]
You updated the version but forgot to change the configuration.

You must now have the document root:
```apache
Alias /ssp /usr/share/self-service-password/htdocs
<Directory /usr/share/self-service-password/htdocs>
AllowOverride None
Require all granted
DirectoryIndex index.php
AddDefaultCharset UTF-8
</Directory>
```

**Reply by ukienitz** [reply]
Thank you very much ! Unfortunately, I now have a blank page. What could be the reason for this?

**Reply by coudot** [reply]
Certainly a missing dependency.

Read https://self-service-password.readthedocs.io/en/v1.4/installation.html

**Reply by ukienitz** [reply]
all dependencies are installed and newest version:
Paket php-Smarty-3.1.33-1.el7.noarch ist bereits in der neuesten Version installiert.
Paket php-common-7.4.27-1.el7.remi.x86_64 ist bereits in der neuesten Version installiert.
Paket php-mbstring-7.4.27-1.el7.remi.x86_64 ist bereits in der neuesten Version installiert.
...

**Reply by ukienitz** [reply]
My Apache Config now is:
Alias /ssp /usr/share/self-service-password/htdocs                                                                           
<Directory /usr/share/self-service-password/htdocs>                                                                          
           AllowOverride None                                                                                                
           Require all granted                                                                                               
           DirectoryIndex index.php                                                                                          
           AddDefaultCharset UTF-8                                                                                           
</Directory>

**Reply by coudot** [reply]
Read the apache error log

**Reply by ukienitz** [reply]
There is no failure....
But the following messages after restart httpd:
[Fri Jan 07 13:21:17.788511 2022] [mpm_prefork:notice] [pid 20819] AH00163: Apache/2.4.52 (codeit) OpenSSL/1.1.1m PHP/7.4.27 configured -- resuming normal operations
[Fri Jan 07 13:21:17.788589 2022] [core:notice] [pid 20819] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND'
[Fri Jan 07 13:31:15.630864 2022] [mpm_prefork:notice] [pid 20819] AH00170: caught SIGWINCH, shutting down gracefully
[Fri Jan 07 13:31:43.796653 2022] [suexec:notice] [pid 22028] AH01232: suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Fri Jan 07 13:31:43.860150 2022] [lbmethod_heartbeat:notice] [pid 22028] AH02282: No slotmem from mod_heartmonitor
[Fri Jan 07 13:31:43.860276 2022] [http2:warn] [pid 22028] AH10034: The mpm module (prefork.c) is not supported by mod_http2. The mpm determines how things are processed in your server. HTTP/2 has more demands in this regard and the currently selected mpm will just not do. This is an advisory warning. Your server will continue to work, but the HTTP/2 protocol will be inactive.
[Fri Jan 07 13:31:43.964864 2022] [mpm_prefork:notice] [pid 22028] AH00163: Apache/2.4.52 (codeit) OpenSSL/1.1.1m PHP/7.4.27 configured -- resuming normal operations
[Fri Jan 07 13:31:43.964929 2022] [core:notice] [pid 22028] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND'

**Reply by coudot** [reply]
Set `$debug = true` in /usr/share/self-service-password/conf/config.inc.php

**Reply by ukienitz** [reply]
There is no failure message... no message:

**Reply by coudot** [reply]
There should be some error message somewhere, configure apche/php/SSP to get it.

If you need professional support, see https://ltb-project.org/professionalservices.html

**Reply by ukienitz** [reply]
Thanks a lot, have a nice weekend!
